### PR TITLE
Adds Sentry `Fingerprint` support

### PR DESCRIPTION
**Pull request by wcharczuk**
Adds fingerprinting support to the Sentry SDK bindings

### CODE REVIEW

File: sentry/client.go [hunk 1]
```diff
@@ -84,6 +85,13 @@ func errEvent(ctx context.Context, ee logger.ErrorEvent) *raven.Event {
 	}
 }
 
+func errFingerprint(ctx context.Context, extra ...string) []string {
+	if fingerprint := GetFingerprint(ctx); fingerprint != nil {
+		return fingerprint
```
Comment (dhermes) [inline sentry/client.go]: OK that `GetFingerprint()` doesn't use the `extra` string slice?

File: sentry/client_test.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ func TestErrEvent(t *testing.T) {
 	assert.Equal(SDK, event.Sdk.Name)
 	assert.Equal("this is a test", event.Message)
 	assert.NotEmpty(event.Exception)
+	assert.NotEmpty(event.Fingerprint)
```
Comment (dhermes) [inline sentry/client_test.go]: Why not check the value?

File: sentry/client_test.go [hunk 3]
```diff
@@ -84,3 +85,63 @@ func TestErrFrames(t *testing.T) {
...
+			Method: "POST",
+			Host:   "example.org",
+			TLS:    &tls.ConnectionState{},
+			URL:    webutil.MustParseURL("https://example.org/foo"),
+		},
+	})
+
+	assert.NotNil(event)
+	assert.NotEmpty(event.Fingerprint)
+	assert.Equal("this is a test", event.Fingerprint[0])
```
Comment (dhermes) [inline sentry/client_test.go]: Nit: I'd prefer

```suggestion
	assert.Equal([]string{"this is a test"}, event.Fingerprint)
```

as future-proofing against that case the `event.Fingerprint` has more than 1 element.

**Comment by dhermes** [comment]
FYI @wcharczuk I rebased this after #161 merged (trying to clean up my own disruption). If your local branch won't push because of "fast forward" issues, that's why.

File: sentry/client_test.go (hunk 2 above)
Comment (wcharczuk) [reply sentry/client_test.go]: there is more detailed test for this later

**Pull request merged**